### DISCUSSION TITLE
「登録されているパターンの一覧が見れる」に関するスレ

### DISCUSSION THREAD

**Discussion by kinakomochigit** [msntts/PatternLanguageEditor#6] [General]
【報告】ComponentとContainer間のデータのやり取りがうまくいかない状態です。今日はこの後すすめれそうにないので、いったん今の段階をpushしました。🙇

**Comment by kinakomochigit** [comment]
@msntts 
パターンのnameの一覧を表示できるようにしました。まずはイメージがあっているか確認お願いします🙇

**Reply by msntts** [reply]
あとで詳しくみてみますが、ぱっと見の実装は良さそう。動かしてみてまだ連絡しますー

**Reply by msntts** [reply]
動作見ました。いい感じだと思います。
後は実装の細かい話を修正する感じかなぁと思います
修正案あれば実装してpushしてみてください。その後、僕のほうで修正してみます。
それを見てもらって感じだったらpull request発行してください🥺

**Reply by kinakomochigit** [reply]
遅くなりましたが、修正案をpushしました。確認よろしくお願いします。🙏

**Reply by msntts** [reply]
自分なりに修正してみました。
なんか結構変更されているように思うかもしれませんが、🍄が作ってくれている状態からの本質的な変更はないです。
修正の観点としては
- モジュールの命名
  - Table=表だけど、今回は一覧表示=ListなのでListに変更
  -  XXComponentとかXXContainerとかそのへんを統一
    - あと、これは今回の開発関係ないけど統一ついでにPattern->PatternViewにした(今後Editくるし)
- モジュールの責務
  - Componentsは言われたものを出す。に集中させたかった
    - 引数から特定のデータを選択する(imgPathを受けるけど使わないとかね)のをやめた
  - selectorを追加して、Contaierがstoreを意識しなくていいようにしてみた(PatternView)は別途storeがいるので未修正

今回の変更を見てみて、よさそうなら(あれば修正してもOK)プルリクエストを出してもらえますか🥺

**Reply by kinakomochigit** [reply]
修正&解説ありがとうございます。おかげで、selectorの使い方が少しわかりました。

**Reply by msntts** [reply]
🎉マージしました🎉
このスレッドをクローズします
